### PR TITLE
Add D18 Hummingbird

### PR DESCRIPTION
**Pull request by FungiFellow**
## About the PR

Added D18 Hummingbird and fixes for the Holdout Pistol. D18 still needs to be mapped in.
## Why / Balance

CM13 Parity, this has no strong effect on Xeno VS Marine gamebalance other than being a Tiny seven 9mm Round Pistol you can hide in your shoes. 

The D18 can be gotten from 4 sources to my knowledge, it is currently unobtainable unless spawned.
1. Loadouts (Unimplemented)
2. Black Market (Unimplemented)
3. Mapped in Spawns (I'm not touching maps quite yet.)
4. Survivors (Unimplemented)

This also corrects the Holdout Pistol's Size and lets it fit in shoes. This will also introduce the .22 cartridge

Resolves https://github.com/RMC-14/RMC-14/issues/3287

- [x] Storage Box
- [x] Crate Info
- [x] Gun Stats
- [x] Magazine
- [x] Sprites
- [x] Aligned Attachment Offsets
- [x] Can slot in Shoes
## Media

image
image
image
image
## Requirements

- [x] I have read and I am following the [Pull Request Guidelines](https://docs.spacestation14.com/en/general-development/codebase-info/pull-request-guidelines.html). I understand that not doing so may get my pr closed at maintainer’s discretion
- [x] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase

## Breaking changes

**Changelog**

:cl:
- add: Added D18 Hummingbird, Holdout Pistol but 9mm
- fix: Fixed Holdout Pistol's ammo type, size and can fit in shoes.

### CODE REVIEW

**Comment by Randomplayer2or3** [comment]
Actually you can get the D18 through Marine loadouts in CM13 if I remember correctly.

**Comment by FungiFellow** [comment]
> Actually you can get the D18 through Marine loadouts in CM13 if I remember correctly.

The code and wiki might just be unclear on it then. I'll ask around.

File: Resources/Prototypes/_RMC14/Entities/Clothing/Shoes/misc.yml [hunk 1]
```diff
@@ -8,6 +8,22 @@
...
+  - type: ItemSlots
+    slots:
+      item:
+        name: rmc-gun
+        insertSound: /Audio/_RMC14/Weapons/Guns/Reload/m42a2.ogg
+        ejectSound: /Audio/_RMC14/Weapons/Guns/Reload/m42a2.ogg
+        whitelist:
+          tags:
+          - Holdout
+  - type: Matchbox
```
Comment (DrSmugleaf) [inline Resources/Prototypes/_RMC14/Entities/Clothing/Shoes/misc.yml:23]: Is this intended? Is this to allow igniting matches with the shoes?

**Review by DrSmugleaf — CHANGES_REQUESTED** [review]
Thanks for the PR!

File: Resources/Prototypes/_RMC14/Entities/Clothing/Shoes/misc.yml (hunk 1 above)
Comment (FungiFellow) [reply Resources/Prototypes/_RMC14/Entities/Clothing/Shoes/misc.yml:23]: That's just leftover from when I copied the section from the Marine Boots. It is possible to use a Shoe's Sole to light a match, so I don't think its malignant at all.

**Comment by FungiFellow** [comment]
Also this will not fit in any non RMC Shoe.

**Comment by FungiFellow** [comment]
Will resolve merge conflicts tomorrow.

File: Resources/Prototypes/_RMC14/Catalog/Cargo/requisitions_catalog.yml [hunk 2]
```diff
@@ -268,3 +268,9 @@
 #      - cost: 3000
 #        crate: RMCCrateMagazineM4SPRCustomIncendiary These ones are straight downgrades and thus shouldn't be for sale
 #      - cost: 4000
+
+
+
+#        Black Market Orders
+#      - cost: 5 W-Y Funnybucks
```
Comment (DrSmugleaf) [inline Resources/Prototypes/_RMC14/Catalog/Cargo/requisitions_catalog.yml:275]:
```suggestion
#      - cost: 5 We-Ya Funnybucks
```

**Review by DrSmugleaf — APPROVED** [review]
Thanks for the PR!

**Pull request merged**